diff --git a/rebuild_remote_test_db.sh b/rebuild_remote_test_db.sh
index fed89e8f531..dd5a1ca2cf0 100755
--- a/rebuild_remote_test_db.sh
+++ b/rebuild_remote_test_db.sh
@@ -3,11 +3,6 @@
 # exit when any command fails
 set -e
 
-DB_USERNAME="DB_USER_FROM_EMAIL"
-DB_PASSWORD="DB_PWD_FROM_EMAIL"
-DB_HOST=sanitized.justworks.com
-DB_NAME=clockwork_test
-
 echo "[ * } DROPPING EXISTING TEST DATABASE"
 mysql --user="${DB_USERNAME}" -h ${DB_HOST} --password="${DB_PASSWORD}" -e "DROP DATABASE IF EXISTS ${DB_NAME}; SHOW DATABASES;"
 
